ivy/functional/frontends/torch/creation_ops.py

```diff
@@ -297,3 +301,32 @@ def zeros_like(
 ):
     ret = ivy.zeros_like(input, dtype=dtype, device=device)
     return ret
+
+
+@with_supported_dtypes({"2.0.1 and below": ("float32", "float64")}, "torch")
+@to_ivy_arrays_and_back
+@handle_out_argument
+def complex(
+    real,
+    imag,
+    *,
+    out=None,
+):
+    assert real.dtype == imag.dtype, ValueError(
+        f"Expected real and imag to have the same dtype, "
+        " but got real.dtype = {real.dtype} and imag.dtype = {imag.dtype}."
+    )
+
+    complex_dtype = ivy.complex64 if real.dtype != ivy.float64 else ivy.complex128
+    complex_array = real + imag * 1j
+    return complex_array.astype(complex_dtype)
```

Comment:
whichever is the last operation you perform, you should use an ivy function for it and then pass `out` argument in it.
So you can do this as following:
`ivy.astype(complex_array, complex_dtype, out=out)`